src/bls12_381/ec.rs

```diff
@@ -1259,13 +1359,27 @@ pub mod g1 {
         assert_eq!(tmp2, c.into_projective());
     }
 
+    #[test]
+    fn test_g1_sw_encode_degenerate() {
+        // test the degenerate cases t = 0 and t^2 = - b - 1
+        let p = G1Affine::sw_encode(Fq::zero());
+        assert!(p.is_on_curve());
+
+        let mut t = Fq::one();
+        t.add_assign(&get_coeff_b());
+        t.negate();
+        let p = G1Affine::sw_encode(t.sqrt().unwrap());
+        assert!(p.is_on_curve());
```

Comment:
1. assert nonzero
2. invoke it with the negative and check that the negative is as expected
3. check the parity of t each time is consistent with the parity of the y-coordinate in the resulting point